packages/payload/src/fields/validations.ts

```diff
@@ -529,11 +529,100 @@ export const array: ArrayFieldValidation = (value, { maxRows, minRows, req: { t
 
 export type BlocksFieldValidation = Validate<unknown, unknown, unknown, BlocksField>
 
+/**
+ * This function validates the blocks in a blocks field against the provided filterOptions.
+ * It will return a list of all block slugs found in the value, the allowed block slugs (if any),
+ * and a list of invalid block slugs that are used despite being disallowed.
+ *
+ * @internal - this may break or be removed at any time
+ */
+export function validateBlocksFilterOptions({
```

Comment:
Can you add an int test for this as well in the accompanying `validations.spec.ts` please?